lib/deaf_grandma.rb

```diff
@@ -21,7 +21,17 @@ def run!
   def speak(input)
 
     #Implement your code here <<<<<<<<<
-
+      if input == "BYE"
+        @bye_counter+=1
+        if @bye_counter == 3
+          @bye_counter = 0
+          "SEE YOU LATER SONNY!"
+        end
+      elsif input.upcase!.nil?
```

Comment:
Avoid modifying the input to the method. Consider:

```ruby
grandma = DeafGrandma.new
speech = "some stuff"
puts grandma.speak(speech) #=> "NOT SINCE 1964!"
puts "I told grandma: #{speech}" #=> "I told grandma: SOME STUFF"
```